SConstruct

```diff
@@ -1135,7 +1139,10 @@ env.Append( CXXFLAGS = [ "-std=$CXXSTD" ] )
 if env["DEBUG"] :
 	env.Append( CXXFLAGS = [ "-g" ] )
 else :
-	env.Append( CXXFLAGS = [ "-DNDEBUG", "-DBOOST_DISABLE_ASSERTS" ] )
+	cxxFlags  =[ "-DNDEBUG", "-DBOOST_DISABLE_ASSERTS"]
+	if env["DEBUGINFO"] :
+		cxxFlags.append( "-g" )
```

Comment:
Does this slow anything down or have any negative impact (e.g would we always want it on, or at least always in the ie config)?